src/Inciter/AMR/refinement.h

```diff
@@ -0,0 +1,779 @@
+            }
+
+            /**
+             * @brief Method which takes a tet id, and deduces the other
+             * parameters needed to perform a 1:2
+             *
+             * @param tet_id The id to refine 1:2
+             */
+            void refine_one_to_two(size_t tet_id)
+            {
+                edge_list_t edge_list = tet_store->generate_edge_keys(tet_id);
+                node_pair_t nodes = find_single_refinement_nodes(edge_list);
+                refine_one_to_two( tet_id, nodes[0], nodes[1]);
+            }
+
+            /**
+             * @brief Method which takes a tet id, and transforms arguments
+             * into the form needed for the main 1:2 refinement method
+             *
+             * @param tet_id The id to refine 1:2
+             */
+            void refine_one_to_two(
+                    size_t tet_id,
+                    std::string edge_key
+            )
+            {
+                std::vector<std::string> nodes = util::split(edge_key,KEY_DELIM);
+                size_t edge_node_A_id =  std::stoul (nodes[0],nullptr,0);
+                size_t edge_node_B_id =  std::stoul (nodes[1],nullptr,0);
+                refine_one_to_two( tet_id, edge_node_A_id, edge_node_B_id);
+            }
+
+            /**
+             * @brief Refine a given tet id into 2 children.
+             * NOTE: Does not do any validity checking (currently?)
+             *
+             * @param tet_id Id of tet to refine
+             * @param edge_node_A_id The first node of id of the edge which
+             * will be split
+             * @param edge_node_B_id The second node of id of the
+             * edge which will be split
+             */
+            void refine_one_to_two(
+                    size_t tet_id,
+                    size_t edge_node_A_id,
+                    size_t edge_node_B_id
+            )
+            {
+
+                if (!check_allowed_refinement(tet_id)) return;
+
+                trace_out << "Refining tet_id " << tet_id << " 1:2 along edge "
+                    << edge_node_A_id << " - " << edge_node_B_id << std::endl;
+
+                //coordinate_t original_tet_c = node_connectivity->id_to_coordinate(id);
+
+                tet_t original_tet = tet_store->get(tet_id);
+
+                // Find midpoint of the edge_id, and make a new node.
+                //coordinate_t mid_point =
+                    //AMR::util::find_mid_point( edge_node_A, edge_node_B);
+
+                size_t new_node_id = node_connectivity->add( edge_node_A_id, edge_node_B_id );
+
+                /// Split existing tet into two new tets
+
+                // The two new tets will be the same, but for each an edge will
+                // be cut, losing an edge  replaced by E
+
+                tet_t new_tet1;
+                tet_t new_tet2;
+
+                // Create a new tet that is based on the original
+                copy_tet(&new_tet1, &original_tet);
```

Comment:
This seems like it would read more naturally with a copy assignment instead. Since `tet_t` is a simple `std::array< std::size_t, 4 >`, this should be done by _not_ defining the copy assignment and the compiler will generate the correct one doing deep-copy as required.